OpenRA.Mods.Common/Traits/World/DomainIndex.cs

```diff
@@ -29,23 +29,22 @@ public void WorldLoaded(World world, WorldRenderer wr)
 		{
 			domainIndexes = new Dictionary<uint, MovementClassDomainIndex>();
 			var tileSet = world.Map.Rules.TileSet;
-			var movementClasses =
-				world.Map.Rules.Actors.Where(ai => ai.Value.HasTraitInfo<MobileInfo>())
-					.Select(ai => (uint)ai.Value.TraitInfo<MobileInfo>().GetMovementClass(tileSet)).Distinct();
+			var locomotors = world.WorldActor.TraitsImplementing<Locomotor>().Where(l => !string.IsNullOrEmpty(l.Info.Name));
+			var movementClasses = locomotors.Select(t => (uint)t.Info.GetMovementClass(tileSet)).Distinct();
 
 			foreach (var mc in movementClasses)
 				domainIndexes[mc] = new MovementClassDomainIndex(world, mc);
 		}
 
-		public bool IsPassable(CPos p1, CPos p2, MobileInfo mi, uint movementClass)
+		public bool IsPassable(CPos p1, CPos p2, LocomotorInfo loco, uint movementClass)
```

Comment:
The movement class is determined from the `LocomotorInfo`.  Do we have a good reason to force callers to pass this in separately?

Reply:
Not as far as I can tell, didn't run into any issues, other than that I need to cache `World` here in DomainIndex now, to be able to pass `world.Map.Rules.TileSet` to `GetMovementClass`.